src/Config/ConfigResolver.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Slim\Console\Config;
+
+use InvalidArgumentException;
+use ReflectionClass;
+use Slim\Console\Exception\ConfigNotFoundException;
+
+use function array_filter;
+use function file_exists;
+use function file_get_contents;
+use function implode;
+use function is_array;
+use function is_null;
+use function json_decode;
+use function strpos;
+
+class ConfigResolver
+{
+    public const FORMAT_JSON = 'json';
+    public const FORMAT_PHP = 'php';
+    //public const FORMAT_YAML = 'yaml';
+    private const CONFIG_FILENAME = 'slim-console.config';
+
+    /** @var string|null */
+    private $file = null;
+
+    /** @var string */
+    private $dir;
+
+    public function __construct(string $dir)
+    {
+        $this->dir = $dir;
+    }
+
+    public function loadConfig(): ?Config
+    {
+        $this->locateConfig();
+
+        return $this->loadConfigByFormat();
+    }
+
+    public function getFile(): string
+    {
+        return $this->file ?? "";
+    }
+
+    protected function locateConfig(): void
+    {
+        $reflection = new ReflectionClass($this);
+        $filterFormats = function ($value, $key) {
+            return strpos($key, 'FORMAT') === 0;
+        };
+        $formats = array_filter($reflection->getConstants(), $filterFormats, ARRAY_FILTER_USE_BOTH);
+
+        foreach ($formats as $possibleFormat) {
+            $fileName = $this->dir . DIRECTORY_SEPARATOR . self::CONFIG_FILENAME . '.' . $possibleFormat;
+            if (file_exists($fileName)) {
+                $this->file = $fileName;
+
+                break;
+            }
+        }
+
+        if (is_null($this->file)) {
+            throw new ConfigNotFoundException(
+                'Please create config file, supported formats: ' . implode(',', $formats)
+            );
+        }
```

Comment:
this `if` statement can be removed since we would reach this from not returning early. The `ConfigurationNotFoundException` still stands though.